### PR TITLE
Upgrade to ocaml-gnt-v2.2.2

### PR DESCRIPTION
**Pull request by andyhhp**
Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>

### CODE REVIEW

**Comment by simonjbeaumont** [comment]
@xen-git retest this please.

**Comment by simonjbeaumont** [comment]
Can we have a changelog entry at the bottom and the release bumped back to 1.

**Comment by simonjbeaumont** [comment]
```
ocamlfind: Package `mirage-profile' not found
W: Field 'pkg_mirage_profile' is not set: Command ''/usr/bin/ocamlfind' query -format %d mirage-profile > '/tmp/oasis-efb62f.txt'' terminated with error code 2
E: Cannot find findlib package mirage-profile
E: Failure("1 configuration error")
make: *** [setup.data] Error 1
...
make[4]: *** [_build/RPMS/x86_64/ocaml-gnt-devel-2.2.2-1.el7.centos.x86_64.rpm] Error 1
```

File: SPECS/ocaml-gnt.spec [hunk 1]
```diff
@@ -1,8 +1,8 @@
 %define debug_package %{nil}
 
 Name:           ocaml-gnt
-Version:        1.0.0
-Release:        2%{?dist}
+Version:        2.2.2
+Release:        1%{?dist}
 Summary:        OCaml bindings for userspace Xen grant table controls
 License:        LGPL2.1 + OCaml linking exception
```
Comment (simonjbeaumont) [inline SPECS/ocaml-gnt.spec]: I see the LICENSE file has dropped off. However I also see from the source tree that the license is no longer LGPL-mumble... It's now ISC: https://github.com/mirage/ocaml-gnt/blob/master/_oasis#L6

**Comment by simonjbeaumont** [comment]
Looking good from a dependency PoV now, just need to fix a wrong license in the repackage of gnt.

File: SPECS/ocaml-gnt.spec [hunk 2]
```diff
@@ -30,6 +31,7 @@ or network backends.
 %package        devel
 Summary:        Development files for %{name}
 Requires:       %{name} = %{version}-%{release}
+Requires:       ocaml-mirage-profile-devel
```
Comment (euanh) [inline SPECS/ocaml-gnt.spec]: All of these Requires: lines for the devel package should have a %{?_isa} suffix.

https://fedoraproject.org/wiki/Packaging:Guidelines#Package_dependencies

File: SPECS/ocaml-gnt.spec [hunk 3]
```diff
@@ -15,6 +15,7 @@ BuildRequires:  ocaml-cstruct-devel
 BuildRequires:  ocaml-io-page-devel
 BuildRequires:  ocaml-lwt-devel
 BuildRequires:  ocaml-cmdliner-devel
+BuildRequires:  ocaml-mirage-profile-devel
 BuildRequires:  xen-devel
 BuildRequires:  xen-dom0-libs-devel
 BuildRequires:  xen-dom0-libs
```
Comment (euanh) [inline SPECS/ocaml-gnt.spec]: This line should not be necessary - xen-dom0-lib-devel should require xen-dom0-libs.

File: SPECS/ocaml-mirage-profile.spec [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+Version:        0.6.1
+Release:        1%{?dist}
+Summary:        Library for collecting trace data
+License:        BSD-2-clause
+URL:            https://github.com/mirage/mirage-profile/
+Source0:        https://github.com/mirage/mirage-profile/archive/v%{version}/%{name}-%{version}.tar.gz
+BuildRequires:  ocaml
+BuildRequires:  ocaml-ocamldoc
+BuildRequires:  ocaml-camlp4-devel
+BuildRequires:  ocaml-findlib
```
Comment (euanh) [inline SPECS/ocaml-mirage-profile.spec]: This line should not be necessary - ocaml-findlib-devel should require ocaml-findlib

File: SPECS/ocaml-gnt.spec (hunk 3 above)
Comment (andyhhp) [inline SPECS/ocaml-gnt.spec]: It does

> %package dom0-libs-devel
> Summary: Xen Hypervisor Domain 0 headers
> Requires: xen-devel = %{version}
> Requires: xen-dom0-libs = %{version}

**Comment by simonjbeaumont** [comment]
@euanh I've removed the reviewed label until you're happy with this. you can add it when you're done with you've finished your round of review.

**Comment by simonjbeaumont** [comment]
:computer: says yes...

**Pull request merged**